plugin/types.go

```diff
@@ -183,24 +191,33 @@ type Streamable interface {
 	Stream(context.Context) (io.ReadCloser, error)
 }
 
-// BlockReadable is an entry with content that can be read in blocks
+// BlockReadable is an entry with data that can be read in blocks.
+// A BlockReadable entry must set its Size attribute. If you don't set it, the
+// file size will be reported as 0 and reads will return an empty file.
 type BlockReadable interface {
 	Entry
 	Read(ctx context.Context, size int64, offset int64) ([]byte, error)
 }
 
-// Readable is an entry with content that can be read
+// Readable is an entry with data that can be read.
 type Readable interface {
 	Entry
 	Read(context.Context) ([]byte, error)
 }
 
-// Writable is an entry that can write data directly to the entry. It mirrors
-// the WriterAt interface with an added context for handling the lifecycle of
-// remote write operations.
+// Writable is an entry that we can write new data to. What that means can be
+// implementation-specific; it could be overwriting a file, submitting a
+// configuration change to an API, or writing data to a queue. It doesn't
+// support a concept of a partial write.
+//
+// Writable can be implemented with or without Readable/BlockReadable. If an
+// entry is only Writable, then only full writes (starting from offset 0) are
+// allowed, anything else initiated by the filesystem will result in an error.
+//
+// It's up to the implementer to decide how much data integrity to guarantee.
```

Comment:
What do you mean by "data integrity"?

Reply:
Primarily around whether writing volumes makes sure an fsync happens so that data is serialized to persistent storage before returning. Not sure how to summarize that.

Reply:
Ah ok. Sounds like an issue for file-like entries, which could be summed up as `plugin.Write(data); data == plugin.Read` (ensure that the next Read from the API will fetch the written data). Don't know if that implies "always call fsync" for volumes.

Reply:
I ended up removing this because I don't think any plugins should explicitly implement something like calling `sync` after writing. It represents a possible loss of data, but the machine would have to die right after and we'd likely get a connection error if that happened. Things like vim and databases will call Fsync, and we're currently lying to them in some cases that it did something, but for now I'd suggest don't run a database on Wash's filesystem.